packages/flutter/lib/src/gestures/binding.dart

```diff
@@ -99,8 +109,60 @@ mixin GestureBinding on BindingBase implements HitTestable, HitTestDispatcher, H
 
   void _flushPointerEventQueue() {
     assert(!locked);
-    while (_pendingPointerEvents.isNotEmpty)
-      _handlePointerEvent(_pendingPointerEvents.removeFirst());
+
+    // Consume all pending pointer events.
+    while (_pendingPointerEvents.isNotEmpty) {
```

Comment:
I am wondering whether we can separate the resampling and the gesture binding a little more, e.g. you only set a PointerResampler on the gesture binding that you want to use and it encapsulates all the logic that's currently in the binding here.

/cc @liyuqian

Reply:
I added a _Resampler helper class to the latest version of this change. It encapsulates the logic to manage multiple PointerEventResamplers and the details of how the scheduler is used to determine sampling time.